scripts/missions/windurst/1_1_The_Horutoto_Ruins_Experiment.lua

```diff
@@ -0,0 +1,393 @@
+-- !addmission 2 0
+--     Gate Guards
+-- Mokyokyo - !pos -55 -8 227 238
+-- Janshura-Rashura - !pos -227 -8 184 240
+-- Rakoh Buuma - !pos 106 -5 -23 241
+-- Zokima-Rokima - !pos 0 -16 124 239
+--
+-- Hakkuru-Rinkuru - !pos -111 -4 101 240
+--
+-- Sama Gohjima - !pos 377 -13 98 116
+--
+-- _5c5 (Gate: Magical Gizmo) - !pos 419 0 -27 192
+--
+--     Gizmos
+-- _5cp (Magical Gizmo) #1 - !pos 464 -3 100 192
+-- _5cq (Magical Gizmo) #2 - !pos 406 -3 59 192
+-- _5cr (Magical Gizmo) #3 - !pos 464 -3 20 192
+-- _5cs (Magical Gizmo) #4 - !pos 295 -3 19 192
+-- _5ct (Magical Gizmo) #5 - !pos 353 -3 60 192
+-- _5cu (Magical Gizmo) #6 - !pos 295 -3 100 192
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/missions')
+require('scripts/globals/npc_util')
+require('scripts/globals/settings')
+require('scripts/globals/keyitems')
+require('scripts/globals/interaction/mission')
+require('scripts/globals/zone')
+-----------------------------------
+local innerHorutotoRuinsID = require("scripts/zones/Inner_Horutoto_Ruins/IDs")
+local eastSarutabarutaID = require("scripts/zones/East_Sarutabaruta/IDs")
+-----------------------------------
+
+local mission = Mission:new(xi.mission.log_id.WINDURST, xi.mission.id.windurst.THE_HORUTOTO_RUINS_EXPERIMENT)
+
+-- A unique title is awarded from accepting the mission at each gate guard
+-- This is handled by zone ID and used in handleAcceptMission below
+local whichTitle =
+{
+    [238] = {titleGiven = xi.title.FRESH_NORTH_WINDS_RECRUIT}, -- Windurst Waters
+    [239] = {titleGiven = xi.title.HEAVENS_TOWER_GATEHOUSE_RECRUIT}, -- Windurst Walls
+    [240] = {titleGiven = xi.title.NEW_BEST_OF_THE_WEST_RECRUIT}, -- Port Windurst
+    [241] = {titleGiven = xi.title.NEW_BUUMAS_BOOMERS_RECRUIT} -- Windurst Woods
+}
+
+mission.reward =
+{
+    rankPoints = 250,
+}
+
+local handleAcceptMission = function(player, csid, option, npc)
+    local zone = player:getZoneID()
+    local which = whichTitle[zone]
+    if option == 1 then
+        mission:begin(player)
+        player:messageSpecial(zones[player:getZoneID()].text.YOU_ACCEPT_THE_MISSION)
+        player:addTitle(which.titleGiven)
+        player:setMissionStatus(mission.areaId, 1)
+    end
+end
+
+local examineGizmo = function(mission, player, gizmoIndex, successCS, failCS)
+    if mission:getVar(player, 'RandomGizmo') == gizmoIndex then
+        return mission:progressEvent(successCS)
+    else
+        if not mission:isVarBitsSet(player, 'GizmoExamined', gizmoIndex) then
+            mission:setVarBit(player, 'GizmoExamined', gizmoIndex)
+            return mission:progressEvent(failCS)
+        else
+            return mission:messageSpecial(innerHorutotoRuinsID.text.EXAMINED_RECEPTACLE)
+        end
+    end
+end
+
+local gizmoSuccess = function(player, csid, option, npc)
+    player:addKeyItem(xi.ki.CRACKED_MANA_ORBS)
+    player:messageSpecial(innerHorutotoRuinsID.text.KEYITEM_OBTAINED, xi.ki.CRACKED_MANA_ORBS)
```

Comment:
These two lines can be replaced by `npcUtil.giveKeyItem(player, xi.ki.CRACKED_MANA_ORBS)`